xarray/core/computation.py

```diff
@@ -969,6 +974,8 @@ def apply_ufunc(
         dimensions as input and vectorize it automatically with
         :py:func:`numpy.vectorize`. This option exists for convenience, but is
         almost always slower than supplying a pre-vectorized function.
+    broadcast : bool
```

Comment:
I think we should skip `apply_ufunc` for now (and just ignore `arithmetic_broadcast` in `dot`)

Seems like there are some confusing implications to implementing it. See https://github.com/pydata/xarray/issues/1618 for an alternative approach.